lib/tasks/convert_embeddables.rake

```diff
@@ -205,98 +176,131 @@ def convert()
+      end
     end
   end
   puts "Deleted #{deleted_embeddables} orphaned ManagedInteractives."
-  # TODO: Delete any LinkedPageItems associated with the deleted ManagedInteractive embeddables.
 end
 
-def delete_old_embeddables()
-  # TODO: Delete all built-in embeddables that are not associated with a PageItem.
-
+def delete_orphaned_old_embeddables()
+  # Delete all built-in embeddables that are not associated with a PageItem.
+  deleted_embeddables = 0
+  @convertable_embeddable_classes.each do |embeddable_class|
+    embeddable_class.constantize.all do |embeddable|
+      if embeddable.page_items.count == 0
+        puts "Deleting orphaned built-in embeddable: #{embeddable.id}"
+        embeddable.destroy
+        deleted_embeddables += 1
+      end
+    end
+  end
+  puts "Deleted #{deleted_embeddables} orphaned old embeddables."
 end
 
 def replace_old_embeddables()
-  # Hook up any ManagedInteractive embeddables that have a legacy_ref_id value and are not 
-  # associated with a PageItem. Using the ManagedInteractive embeddable's legacy_ref_id,
-  # find the built-in embeddable and PageItem that it is associated with. Then, replace the 
-  # PageItem's built-in embeddable with the ManagedInteractive embeddable. (Delete the old, 
-  # built-in embeddable?) We will also need to add LinkedPageItems to
-  ManagedInteractive.where("legacy_ref_id IS NOT NULL AND legacy_ref_id != ?", "").each do |mi|
-    if PageItem.where(:embeddable_id => mi.id, :embeddable_type => "ManagedInteractive").count == 0
-      old_embeddable_id, old_embeddable_type = mi.legacy_ref_id.split("-")
-      page_item = PageItem.where(:embeddable_id => old_embeddable_id, :embeddable_type => old_embeddable_type).first
-      if page_item
-        puts "page_item: #{page_item.inspect}"
-        # TODO: Update the page item's embeddable to be the ManagedInteractive embeddable.
+  # To replace the old, built-in embeddables, first we find any ManagedInteractive embeddables
+  # that have a legacy_ref_id value and are not yet associated with a PageItem. Then, we use 
+  # each of these ManagedInteractive embeddables' legacy_ref_id to find the old embeddable it 
+  # will replace and the PageItem that it's associated with. We replace the PageItem's old 
+  # embeddable with the new embeddable. We also update any wrapping plugins for the old embeddable
+  # so they point to the new embeddable. Finally, delete the old embeddable.
+  new_embeddables = ManagedInteractive.where("legacy_ref_id IS NOT NULL AND legacy_ref_id != ?", "")
+  new_embeddables.each do |new_embeddable|
+    if PageItem.where(:embeddable_id => new_embeddable.id, :embeddable_type => "ManagedInteractive").count == 0
+      old_embeddable_id, old_embeddable_type = new_embeddable.legacy_ref_id.split("-")
+      old_embeddable = old_embeddable_type.constantize.find(old_embeddable_id)
+      page_items = PageItem.where(:embeddable_id => old_embeddable.id, :embeddable_type => old_embeddable_type)
```

Comment:
This will also result in a lot of queries, but I don't see a way to avoid that unless we stored additional fields on the ManagedInteractive that made it possible possible to do a SQL join with the legacy page item. If this is too slow it might be worth doing that.